tortoisebeacon/tortoise_beacon.go

```diff
@@ -503,84 +503,61 @@ func (tb *TortoiseBeacon) runConsensusPhase(ctx context.Context, epoch types.Epo
 	// For next rounds,
 	// wait for δ time, and construct a message that points to all messages from previous round received by δ.
 	// rounds 1 to K
-	timer := time.NewTimer(tb.config.FirstVotingRoundDuration + tb.config.WeakCoinRoundDuration)
+	timer := time.NewTimer(tb.config.FirstVotingRoundDuration)
 	defer timer.Stop()
 
 	var (
-		coinFlip            bool
-		ownLastRoundVotesMu sync.RWMutex
-		ownLastRoundVotes   allVotes
+		ownVotes allVotes
+		err      error
 	)
-
 	for round := firstRound; round <= tb.lastRound(); round++ {
-		// always use coinflip from the previous round for current round.
-		// round 1 is running without coinflip (e.g. value is false) intentionally
 		round := round
-		previousCoinFlip := coinFlip
+		votes := ownVotes
 		tb.eg.Go(func() error {
 			if round == firstRound {
 				if err := tb.sendProposalVote(ctx, epoch); err != nil {
```

Comment:
sendProposalVote() should initialize ownVotes

Reply:
why? in current code votes are initialized by calcVotes after received all proposals. it is not correct?

Reply:
i think sendProposalVote will just use initial votes for first voting message that were formed from all proposals

Reply:
can you point me to the code where this node's own proposal is added to votesMargin? i am lost in the sea of code now...

Reply:
> why? in current code votes are initialized by calcVotes after received all first round messages. it is not correct?
it is, but that's only counting votes the node received. i cannot find where this node's own initial proposal/vote was added.

Reply:
https://github.com/spacemeshos/go-spacemesh/blob/782b11dcdd98ee10c4c43375f82a6236a192715c/tortoisebeacon/tortoise_beacon.go#L486-L489

Reply:
@dshulyak thanks. but this only serves to send our own proposal to peers, and in later encoding for following votes.
i still don't see where `tb.votesMargin` is updated with own initial votes.
`tb.votesMargin` seems to be only updated when it receives a gossip message from peers.

Reply:
ah sorry i understood now what we were talking about. everything that is Broadcast'ed will be sent to the local handler for the protocol as well. So, what is sent in `sendProposalVote` will be also received HandleSerializedFirstVotingMessage. that's sort of a feature of p2p framework

Reply:
OMG... i'm learning something new about p2p every day. cool! thanks for your patience @dshulyak

Reply:
but then, does that mean we add our own proposal to the valid list twice? once in `proposalPhaseImpl` and once in the proposal handler? @dshulyak

Reply:
i think this is the case. i don't see any code that prevents adding duplicate proposals